packages/blade/src/components/Table/_decisions/decisions.md

```diff
@@ -0,0 +1,541 @@
+- Replicating the Composable API would be a challenging task with React Table since it internally uses API similar to the Compact API. We would have to build a lot of abstractions to ensure we can build the Composable API on top of the Compact API.
+- React Table is possibly our 2nd best option if we ever need more complex tables in the future but as of now it would be an overkill for us
+
+#### Small sized library - React Table Library
+- [React Table Library (@table-library/react-table-library)](https://github.com/table-library/react-table-library) is the best example of a small sized library. It has all the features we need and is relatively well maintained.
+- It contains all the features we need today as well as the ones we can forsee needing in the future
+- As of today, React Table Library's bundle size is 23kb (7kb gzipped).
+- React Table Library has a Composable API using which we can build our Composable as well as Compact API
+- React Table Library also exposes a minimal Compact API but we cannot use that directly to build our Compact API since there isn't a parity between their own Composable & Compact APIs
+- React Table Library is our best option as of today since it has all the features we could need and is relatively lightweight
+
+#### Comparison
+| Library                                                                     | Features                                                      | Bundle Size                                             | Composable API       | Compact API   | Maintenance                                             | Our Choice |
+| --------------------------------------------------------------------------- | ------------------------------------------------------------- | ------------------------------------------------------- | -------------------- | ------------- | ------------------------------------------------------- | ---------- |
+| [AG Grid](https://www.ag-grid.com/)                                         | Very heavy and has a lot of features we don't need            | 75kb (15kb gzipped) (+ additional supporting libraries) | Challenging to build | Easy to build | Extremely well maintained. Has paid Enterprise support. |
+| [Tanstack's React Table](https://react-table.tanstack.com/)                 | Relatively heavy and has a lot of features we don't need      | 60kb (14kb gzipped)                                     | Challenging to build | Easy to build | Very well maintained                                    |
+| [React Table Library](https://github.com/table-library/react-table-library) | Relatively lightweight and has all the features we could need | 23kb (7kb gzipped)                                      | Easy to build        | Easy to build | Relatively well maintained                              | ✅          |
```

Comment:
What about tables for native? Can we use the same libraries? Any other things that may change (eg. virtualization)

Reply:
We're scoping Table only for web & mweb. Native has lists instead of tables. Will add this to the doc